### ISSUE TITLE
[BUG] Error when first playing a video with software decoder and then switching directly to a video with hardware decoder.

### ISSUE THREAD

**Issue by ismail-38** [anilbeesetti/nextplayer#699]
**Describe the bug**
The player stops playing and displays an error if a video with a software decoder (ffmpeg) is played first, and then a video with a hardware decoder is to be played via the "Skip To Next" button. 
ExtensionRendererMode = EXTENSION_RENDERER_MODE_ON

**To Reproduce**
1. Put 2 video files in a directory, 1 should need a software decoder.
2. Select the decoder priority -> Prefer device decoder
3. Start 1 video, and switch back and forth between the two videos.

**Expected behavior**
After using a software decoder, the initialization of the hardware decoder should work. Thus the problem would be solved.

**Screenshots**

**Device info (please complete the following information):**
 - Device: Samsung S23 Ultra
 - Android version: 13
 - App version: 0.9.0

**Additional context**
Workaround: 
Close the player and call it again. Since the player instance is created again, it works again. But of course it is unattractive. 

Thoughts:
- Does the software decoder still hold any resources, which is why the hardware decoder cannot be initialized?
- what I also haven't understood yet is (would have to debug), why in the method "MediaCodecRenderer.maybeInitCodecWithFallback()" it doesn't fall back to the software renderer, if the hardware renderer can't be created. Possibly another bug in Media3. But the original problem is probably in NextLib.

**Comment by odkrys** [comment]
This is the bug I experienced. After MPEG2 DVD video, playback fails for hardware codecs such as H264, HEVC, VP9, ​​and AV1. There is a problem with the transition from software decoding to hardware decoding.

It works if I generate a PlayerSurface every time the media ID changes and reconnect in the playerservice on an onPlayerError, but I am not sure if that is a good solution.

**Comment by samaudi** [comment]
Yes, the problem is in NextLib; see also [https://github.com/anilbeesetti/nextlib/issues/190](https://github.com/anilbeesetti/nextlib/issues/190)

**Comment by anilbeesetti** [comment]
Fixed a while back, now this won't happen. If it does please reopen the issue
